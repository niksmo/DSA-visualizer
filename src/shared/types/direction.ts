export const enum Direction {
  Ascending = 'ascending',
  Descending = 'descending'
}
