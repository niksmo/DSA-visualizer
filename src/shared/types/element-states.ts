export const enum ElementStates {
	Default = 'default',
	Changing = 'changing',
	Modified = 'modified'
}
