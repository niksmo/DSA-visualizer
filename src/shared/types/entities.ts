import { ElementStates } from './element-states';

export type TArrayItem<T = number> = {
  id: string;
  value: T;
  state: ElementStates;
  head?: string | null;
  tail?: string | null;
  passed?: boolean;
};

export type TListNode<T> = {
  id: string;
  value: T;
  state: ElementStates;
  next: TListNode<T> | null;
};
