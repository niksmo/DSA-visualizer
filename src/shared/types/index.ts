import { Direction } from './direction';
import { ElementStates } from './element-states';
import { TArrayItem, TListNode } from './entities';

export { Direction, ElementStates };
export type { TArrayItem, TListNode };
