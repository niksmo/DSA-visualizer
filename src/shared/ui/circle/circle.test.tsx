import { render } from '@testing-library/react';
import { ElementStates } from 'types';
import { Circle } from '.';

describe('Circle component test', () => {
  it('should render with text', () => {
    const { getByTestId } = render(<Circle letter="abcd" />);

    const circleWithText = getByTestId('circle');

    expect(circleWithText).toHaveTextContent('abcd');

    expect(circleWithText).toMatchSnapshot();
  });

  it('should render without text', () => {
    const { getByTestId } = render(<Circle />);

    const emptyCircle = getByTestId('circle');

    expect(emptyCircle).toBeInTheDocument();

    expect(emptyCircle).toMatchSnapshot();
  });

  it('should render with text in head', () => {
    const { getByTestId } = render(<Circle head="abcd" />);

    const circleWithTextInHead = getByTestId('circle');

    expect(circleWithTextInHead).toHaveTextContent('abcd');

    expect(circleWithTextInHead).toMatchSnapshot();
  });

  it('should render with react element in head', () => {
    const { getByTestId } = render(<Circle head={<a>abcd</a>} />);

    const circleWithElemInHead = getByTestId('circle');

    expect(circleWithElemInHead).toHaveTextContent('abcd');

    expect(circleWithElemInHead).toMatchSnapshot();
  });

  it('should render with text in tail', () => {
    const { getByTestId } = render(<Circle tail="abcd" />);

    const circleWithTextInTail = getByTestId('circle');

    expect(circleWithTextInTail).toHaveTextContent('abcd');

    expect(circleWithTextInTail).toMatchSnapshot();
  });

  it('should render with react element in tail', () => {
    const { getByTestId } = render(<Circle tail={<a>abcd</a>} />);

    const circleWithElemInTail = getByTestId('circle');

    expect(circleWithElemInTail).toHaveTextContent('abcd');

    expect(circleWithElemInTail).toMatchSnapshot();
  });

  it('should render with index', () => {
    const { getByTestId } = render(<Circle index={5} />);

    const circleWithIndex = getByTestId('circle');

    expect(circleWithIndex).toHaveTextContent('5');

    expect(circleWithIndex).toMatchSnapshot();
  });

  it('should render with isSmall prop', () => {
    const { getByTestId } = render(<Circle isSmall />);

    const circleWithIndex = getByTestId('circle-main');

    expect(circleWithIndex).toHaveClass('small');

    expect(circleWithIndex).toMatchSnapshot();
  });

  it('should render with all existing states', () => {
    const { container, getByText } = render(
      <>
        <Circle state={ElementStates.Default} letter="1" />
        <Circle state={ElementStates.Changing} letter="2" />
        <Circle state={ElementStates.Modified} letter="3" />
      </>
    );

    const defaultState = getByText('1');
    const changingState = getByText('2');
    const modifiedState = getByText('3');

    expect(defaultState).toHaveTextContent('1');
    expect(changingState).toHaveTextContent('2');
    expect(modifiedState).toHaveTextContent('3');

    expect(container).toMatchSnapshot();
  });
});
