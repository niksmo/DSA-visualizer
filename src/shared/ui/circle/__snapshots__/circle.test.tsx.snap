// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component test should render with all existing states 1`] = `
<div>
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   default"
      data-testid="circle-main"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="circle-value"
      >
        1
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
      data-testid="circle-index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   changing"
      data-testid="circle-main"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="circle-value"
      >
        2
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
      data-testid="circle-index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
  <div
    class="content "
    data-testid="circle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
      data-testid="circle-head"
    />
    <div
      class="circle   modified"
      data-testid="circle-main"
    >
      <p
        class="text text_type_circle text_color_input letter"
        data-testid="circle-value"
      >
        3
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
      data-testid="circle-index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
      data-testid="circle-tail"
    />
  </div>
</div>
`;

exports[`Circle component test should render with index 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  >
    5
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`Circle component test should render with isSmall prop 1`] = `
<div
  class="circle  small default"
  data-testid="circle-main"
>
  <p
    class="text text_type_circle text_color_input letter"
    data-testid="circle-value"
  />
</div>
`;

exports[`Circle component test should render with react element in head 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  >
    <a>
      abcd
    </a>
  </div>
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`Circle component test should render with react element in tail 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  >
    <a>
      abcd
    </a>
  </div>
</div>
`;

exports[`Circle component test should render with text 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-value"
    >
      abcd
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`Circle component test should render with text in head 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
    data-testid="circle-head"
  >
    abcd
  </div>
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;

exports[`Circle component test should render with text in tail 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
    data-testid="circle-tail"
  >
    abcd
  </div>
</div>
`;

exports[`Circle component test should render without text 1`] = `
<div
  class="content "
  data-testid="circle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
    data-testid="circle-head"
  />
  <div
    class="circle   default"
    data-testid="circle-main"
  >
    <p
      class="text text_type_circle text_color_input letter"
      data-testid="circle-value"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
    data-testid="circle-index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
    data-testid="circle-tail"
  />
</div>
`;
